### PR TITLE
refactor: resolve sonarcloud bugs

### PR DESCRIPTION
**Pull request by Souvikns**
**Description**
This PR aims to resolve all the bugs and code smells thrown by Sonarcloud. 

**Related issue(s)**

Fixing it for #430

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -85,62 +85,62 @@ export default async function GleeAppInitializer () {
...
   })
   
-  app.on('adapter:reconnect', (e: EnrichedEvent) => {
+  app.on('adapter:reconnect', async (e: EnrichedEvent) => {
     logLineWithIcon('↪', `Reconnected to server ${e.serverName}.`, {
       highlightedWords: [e.serverName],
       iconColor: '#0f0',
     })
-    runLifecycleEvents('onReconnect', {
+    await runLifecycleEvents('onReconnect', {
```
Comment (KhudaDad414) [inline src/index.ts:104]: I think we should set the `captureRejections` to `true` in `glee.ts` constructor and have a `on('error')` to capture the errors we can handle the rejections here.

File: src/lib/glee.ts [hunk 2]
```diff
@@ -37,8 +37,10 @@ export default class Glee extends EventEmitter {
    *
    * @param {Object} [options={}]
    */
-  constructor (options:GleeConfig = {}) {
-    super()
+  constructor(options: GleeConfig = {}) {
+    super({ captureRejections: true })
+
+    this.on('error', (error) => console.log(error))
```
Comment (Souvikns) [inline src/lib/glee.ts:43]: @KhudaDad414 let me know if this will do, for now, just logging the error right now.

Comment (KhudaDad414) [inline src/lib/glee.ts:43]:
```suggestion
    this.on('error', console.error)
```
just a little easier to read.

**Comment by KhudaDad414** [comment]
@Souvikns tests are failing for some weird reason 🤔

**Comment by Souvikns** [comment]
> @Souvikns tests are failing for some weird reason 🤔

I think @jimp/plugin-rotate@0.14.0 for removed or something.

**Comment by Souvikns** [comment]
@KhudaDad414 need your review on this.

**Review by KhudaDad414 — APPROVED** [review]
Looks good. :)

**Pull request merged**